cpp/include/cudf/copying.hpp

```diff
@@ -536,11 +549,42 @@ struct contiguous_split_result {
  * @return The set of requested views of `input` indicated by the `splits` and the viewed memory
  * buffer.
  */
-std::vector<contiguous_split_result> contiguous_split(
+std::vector<packed_table> contiguous_split(
   cudf::table_view const& input,
   std::vector<size_type> const& splits,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Deep-copy a `table_view` into a serialized contiguous memory format
+ *
+ * The metadata from the `table_view` is copied into a host vector of bytes and the data from the
+ * `table_view` is copied into a `device_buffer`. Pass the output of this function into
+ * `cudf::unpack` to deserialize.
+ *
+ * @param input View of the table to pack
+ * @param[in] mr Optional, The resource to use for all returned device allocations
+ * @return packed_columns A struct containing the serialized metadata and data in contiguous host
+ *         and device memory respectively
+ */
+packed_columns pack(cudf::table_view const& input,
+                    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief Deserialize the result of `cudf::pack`
+ *
+ * Converts the result of a serialized table into a `table_view` that points to the data stored in
+ * the contiguous device buffer contained in `input`.
+ *
+ * It is the caller's responsibility to ensure that the `table_view` in the output does not outlive
+ * the data in the input.
+ *
+ * No new device memory is allocated in this function.
+ *
+ * @param input The packed columns to unpack
+ * @return The unpacked `table_view`
+ */
+table_view unpack(packed_columns const& input);
```

Comment:
I'd like to see a way to unpack the data that doesn't involve having a `packed_columns` structure to pass as input.  The problem is that `packed_columns` owns the data, both the metadata because it's a `std::vector` and the table data because it's an `rmm::device_buffer`.

It's likely we will ship the metadata across the wire as part of some larger message, and it'd be nice to avoid the requirement to copy the metadata out into a separate `std::vector` in order to call `unpack`.  Similarly there are cases where we will pack multiple tables together into a single transfer message, and it'd be nice to not have to copy the GPU data to build a temporary view of the packed table.

Reply:
I can add:

`table_view unpack(uint8_t const* metadata, uint8_t const* gpu_data)`